test/io/mandoline/impl_test.clj

```diff
@@ -33,3 +37,34 @@
       ;; Even though we were writing thousands of chunks, they were all the same,
       ;; and hopefully we didn't write that one chunk too many times.
       (is (< 0 @chunks-written 100)))))
+
+(deftest test-bytes-to-array
+  (let [elements (map byte (range 30))
+        ; Create a ByteBuffer instance that is backed by a 25-element
+        ; byte array, with position=3 and limit=19. Diagram:
+        ;
+        ; byte[25] =========================
+        ; slice        =====================
+        ; limit        ===================
+        ; position        ================
+        byte-buffer (-> (byte-array elements)
+                      (ByteBuffer/wrap)
+                      (.position 4) ; advance position by 4
+                      (.slice) ; create a slice that shares the underlying byte[]
+                      (.limit 19) ; trim to 19 long
+                      (.position 3)) ; advance position by 3
+        data-type DataType/BYTE
+        shape [2 4 2]
+        array (impl/bytes-to-array byte-buffer data-type shape)]
+    (is (instance? Array array)
+        "bytes-to-array returns an instance of ucar.ma2.Array")
+    (is (= (.getClassType data-type) (.getElementType array))
+        "bytes-to-array returns an Array with the expected element type")
+    (is (= shape (vec (.getShape array)))
+        "bytes-to-array returns an Array with the expected shape")
+    (is (= (take (- 19 3) (drop (+ 4 3) elements)) (seq (.copyTo1DJavaArray array)))
+        "bytes-to-array returns an Array that contains the expected elements")
+    (is (= 3 (.position byte-buffer))
+        "bytes-to-array returns the ByteBuffer with its original position")
```

Comment:
nit: It may be more proper to say "bytes-to-array does not modify the original position of the ByteBuffer". Same applies to the limit test below.